internal/controllers/s3userclaim/provisioner.go

```diff
@@ -131,6 +133,81 @@ func (r *Reconciler) ensureReadonlySubuser(ctx context.Context) (*ctrl.Result, e
 	return subreconciler.ContinueReconciling()
 }
 
+func (r *Reconciler) ensureOtherSubusers(ctx context.Context) (*ctrl.Result, error) {
+	specSubUsers := r.s3UserClaim.Spec.SubUsers
+	// Create a hashmap to move all spec and ceph subUsers to it
+	subUserFullIdSet := make(map[string]string)
+
+	// Tag specSubUsers with "create"
+	for _, subUser := range specSubUsers {
+		cephSubUserFullId := cephSubUserFullIdMaker(r.cephUserFullId, subUser)
+		subUserFullIdSet[cephSubUserFullId] = consts.SubUserTagCreate
+	}
+
+	// Add read-only subUser to subUsers to prevent if from removing
+	subUserFullIdSet[r.readonlyCephUserFullId] = consts.SubUserTagCreate
+
+	// Tag cephSubUsers as remove if they are not already in the hashmap and remove them otherwise
+	// since they are already available on ceph and not needed to created.
+	for _, cephsubUser := range r.cephUser.Subusers {
+		_, exists := subUserFullIdSet[cephsubUser.Name]
+		if exists {
+			delete(subUserFullIdSet, cephsubUser.Name)
+		} else {
+			subUserFullIdSet[cephsubUser.Name] = consts.SubUserTagRemove
+		}
+	}
+
+	// Iterate over the subUsers hashmap and create or remove subUsers according to their tags.
+	for subUserFullId, tag := range subUserFullIdSet {
+		desiredSubuser := admin.SubuserSpec{
+			Name:    subUserFullId,
+			Access:  admin.SubuserAccessNone,
+			KeyType: pointer.String(consts.CephKeyTypeS3),
+		}
+		if tag == consts.SubUserTagCreate {
+			// Create the subuser
+			r.logger.Info(fmt.Sprintf("Create subUser: %s", subUserFullId))
+			if err := r.rgwClient.CreateSubuser(ctx, admin.User{ID: r.cephUserFullId}, desiredSubuser); err != nil {
+				r.logger.Error(err, "failed to create subUser")
+				return subreconciler.Requeue()
+			}
+		} else {
+			// Delete the subuser
```

Comment:
Unnecessary comment. (Also the previous comment `Create the subuser`)

Reply:
Done